forge-gui/res/cardsfolder/upcoming/wreck_and_rebuild.txt

```diff
@@ -0,0 +1,9 @@
+Name:Wreck and Rebuild
+ManaCost:1 R G
+Types:Sorcery
+A:SP$ Charm | CharmNum$ 1 | Choices$ DestroyArtifact,DBMill
+SVar:DestroyArtifact:DB$ Destroy | ValidTgts$ Artifact,Enchantment | TgtPrompt$ Select target artifact or enchantment | SpellDescription$ Destroy target artifact or enchantment.
+SVar:DBMill:DB$ Mill | NumCards$ 3 | Defined$ You | SubAbility$ DBReturn | SpellDescription$ Mill five cards, then you may put a land card from your graveyard onto the battlefield tapped.
+SVar:DBReturn:DB$ ChangeZone | Origin$ Graveyard | Destination$ Battlefield | Optional$ True | ChangeType$ Land.YouOwn | Hidden$ True | Tapped$ True
```

Comment:
ChangeZone doesn't use Optional param because it is optional by default

ChangeZone requires Mandatory param when it isn't optional

Reply:
thanks for the explanation !